client/src/components/activities/ActivityForm/useActivityForm.ts

```diff
@@ -158,13 +113,121 @@ export default function useActivityForm({
 			} as WithDates | WithTimestamps)
 		: undefined;
 
+	// stuff that was previously in useRecurrenceForm
+
+	const [isRecurring, setIsRecurring] = useState(false);
+
+	// TODO (TRK-204): remove this
+	useEffect(() => {
+		console.log({ recurrence });
+	}, [recurrence]);
+
+	function resetSelection() {
+		setRecurrence(
+			produce((draft) => {
+				draft.weekdays = null;
+				draft.monthdays = null;
+			})
+		);
+	}
+
+	function setSelection<T extends SetRecurrenceSelection["type"]>(type: T) {
+		return (value: Extract<SetRecurrenceSelection, { type: T }>["value"] | null) =>
+			setRecurrence(
+				produce((draft) => {
+					// The logic for these cases is basically the same, but the
+					// typing is different, so it's easier to write it out twice.
+					if (type === "weekdays") {
+						const v = value as DayOfWeek;
+						draft.monthdays = null;
+
+						draft.weekdays ??= [];
+						if (draft.weekdays.includes(v)) {
+							draft.weekdays.splice(draft.weekdays.indexOf(v), 1);
+						} else {
+							draft.weekdays.push(v);
+						}
+					} else {
+						const v = value as number;
+						draft.monthdays ??= [];
+
+						if (draft.monthdays.includes(v)) {
+							draft.monthdays.splice(draft.monthdays.indexOf(v), 1);
+						} else {
+							draft.monthdays.push(v);
+						}
+
+						draft.weekdays = null;
+					}
+				})
+			);
+	}
+
+	function toggleRecurring() {
```

Comment:
Rename the recurrence-specific functions to make it clear they relate to recurrence. For example, `resetSelection` isn't very verbose.